.github/workflows/update-translations.yml

```diff
@@ -1,22 +1,56 @@
 name: Update translations from Weblate
+permissions:
+  contents: write
+  pull-requests: write
 on:
+  pull_request:
   schedule:
     - cron: "0 10 * * 1"
   workflow_dispatch:
+
 jobs:
   update-translations:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
       - name: Fetch and merge from Weblate
         # The commit message is generated in Weblate; see https://hosted.weblate.org/addon/17163/
         run: |
+          git checkout -b main
+          git log --oneline --graph --all -n5
           git remote add weblate https://hosted.weblate.org/git/zulip/zulip-flutter/
           git fetch weblate
-          git merge --ff-only weblate/main
+          GIT_COMMITTER_NAME="Hosted Weblate" GIT_COMMITTER_EMAIL="hosted@weblate.org" \
+            git cherry-pick weblate/main ^HEAD
```

Comment:
Hmm I guess there's a race where if origin/main just got updated and Weblate hasn't taken that update yet, the `merge --ff-only` will fail.

How about `reset --hard`? Then the PR's base will be slightly behind the tip, but that's normal. We'll rebase it forward as part of merge like we do for many other PRs.

Reply:
Ah I see, this change is part of the "run on this PR" don't-merge commit. Never mind, then.

(There still is that race, but it's unlikely.)

Reply:
One option is we can `git reset --hard weblate/main` and let the PR potentially be behind `main` if weblate is lagging.  Which means one might get a PR with "cannot merge" rather than the failure one gets with `git merge --ff-only`, which is that the whole action fails.

Reply:
Yeah, I think lagging behind `main` is fine. We can always re-run the workflow.